x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
this whole file looks like something that could be abstracted away ? for visualizations saved object i mostly just copy this and rename Map to Visualization.

Reply:
Creating abstraction for multiple teams to consume is out of scope for this PR. This PR is to show the way to onboard maps (single types). If teams who own multiple SO types want to create their abstraction they can do it freely.